velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -383,179 +383,383 @@ class SharedArbitrationTest : public exec::test::HiveConnectorTestBase {
-        .spillDirectory(spillDirectory->path)
-        .config(core::QueryConfig::kSpillEnabled, "true")
-        .config(core::QueryConfig::kTopNRowNumberSpillEnabled, "true")
-        .queryCtx(queryCtx)
-        .maxDrivers(numDrivers)
-        .plan(PlanBuilder(planNodeIdGenerator)
-                  .values(vectors)
-                  .project({"c1"})
-                  .topN({"c1 NULLS FIRST"}, 10, false)
-                  .planNode())
-        .assertResults(
-            "SELECT c1 "
-            "FROM tmp "
-            "ORDER BY c1 ASC NULLS LAST "
-            "LIMIT 10");
+  core::PlanNodePtr writePlan(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::string& outputDirPath) {
+    return PlanBuilder()
+        .values(vectors)
+        .tableWrite(outputDirPath)
+        .singleAggregation(
+            {},
+            {fmt::format("sum({})", TableWriteTraits::rowCountColumnName())})
+        .planNode();
+  }
+
+  RowVectorPtr runWriteTask(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::shared_ptr<core::QueryCtx>& queryCtx,
+      uint32_t numDrivers,
+      bool enableSpilling) {
+    const auto outputDirectory = TempDirectoryPath::create();
+    auto plan = writePlan(vectors, outputDirectory->path);
+    if (enableSpilling) {
+      const auto spillDirectory = exec::test::TempDirectoryPath::create();
+      return AssertQueryBuilder(plan)
+          .spillDirectory(spillDirectory->path)
+          .config(core::QueryConfig::kSpillEnabled, "true")
+          .config(core::QueryConfig::kWriterSpillEnabled, "true")
+          // Set 0 file writer flush threshold to always trigger flush in
+          // test.
+          .config(
+              core::QueryConfig::kWriterFlushThresholdBytes,
+              folly::to<std::string>(0))
+          // Set stripe size to extreme large to avoid writer internal
+          // triggered flush.
+          .connectorConfig(
+              kHiveConnectorId,
+              connector::hive::HiveConfig::kOrcWriterMaxStripeSize,
+              folly::to<std::string>("1GB"))
+          .connectorConfig(
+              kHiveConnectorId,
+              connector::hive::HiveConfig::kOrcWriterMaxDictionaryMemory,
+              folly::to<std::string>("1GB"))
+          .queryCtx(queryCtx)
+          .maxDrivers(numDrivers)
+          .copyResults(pool());
+    } else {
+      return AssertQueryBuilder(plan)
+          .queryCtx(queryCtx)
+          .maxDrivers(numDrivers)
+          .copyResults(pool());
+    }
   }
 
   std::shared_ptr<Task> runWriteTask(
       const std::vector<RowVectorPtr>& vectors,
       const std::shared_ptr<core::QueryCtx>& queryCtx,
-      uint32_t numDrivers) {
-    int numRows = std::accumulate(
-        vectors.cbegin(),
-        vectors.cend(),
-        0,
-        [](int sum, const RowVectorPtr& row) { return sum + row->size(); });
-    const auto spillDirectory = exec::test::TempDirectoryPath::create();
+      uint32_t numDrivers,
```

Comment:
There is a good amount of copy-paste between 2 runWriteTask methods. It would be nice to refactor to avoid code repeat. Maybe,

```
   auto result = runWriteTask(...);
   assertEqualResults(result, expectedResult);
```

Reply:
Good point.